### ISSUE TITLE
Indexed properties in combination with an iframe 

### ISSUE THREAD

**Issue by Bella1977** [cheezy/page-object#343]
Hello, 
I hope for someone can help me with this request. 

I am wondering whether the indexed properties are also supported when they are nested in an iframe. 
in_iframe(:id => ReportFrame') do |frame|
    indexed_property(:chain_list,[
                         [:checkbox,  :select_chain, {:id => 'iclChain_rdList%s'}, :frame => frame]
    ])
end
And unfortunately with this I get the error message, that the element can't be found. 
Sandra

**Comment by Th33x1l3** [comment]
Same problem here. I even tried the following:
```

indexed_property(:user_row_contents, [
          [:row, :user_row, {:id => 'linhas_corpo_tabela_%s',:frame=> in_frame}]])
```

and i've got the following:

`[WARN] @ 2016-07-27 11:29:37 : undefined method `%' for [{:frame=>{:name=>"mainFrame_geral"}}]:Array`

**Comment by vietch2612** [comment]
'linhas_corpo_tabela_%s' -> "linhas_corpo_tabela_%s"

You should use "".

**Comment by Th33x1l3** [comment]
I may have a possible solution, just don't know how to commit it with tests:
in page-object/indexed-properties.rb
```
def initialize (browser, index, identifier_list)
            initialize_browser(browser)

            identifier_list.each do |identifier|
              type = identifier[0]
              name = identifier[1]
              how_and_what = identifier[2].clone # Cannot modify the original...
              how_and_what.each do |key, value|
                case key
                  when :index
                    how_and_what[key] = (value % index).to_i
                  when :frame
                    how_and_what[key] = value # don't substitute frame
                  else
                    how_and_what[key] = value % index
                end
              end
              self.class.send type, name, how_and_what unless Class.instance_methods.include? name
            end
          end
```

i just added the :frame case. 

This patch uses the syntax

`[:<type>, :<name>, {<type_identifier>: '<identifier_str>', frame: <frame_to_be_used>}]`

**Comment by AlexisKAndersen** [comment]
This issue is being addressed and will be fixed in the upcoming 2.2.0 release.
